File: Backend/models/Provider.js
import mongoose from "mongoose";

const providerSchema = new mongoose.Schema({
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: "User",
    required: true,
    unique: true,
  },
  businessName: {
    type: String,
    required: true,
  },
  description: {
    type: String,
    required: true,
  },
  category: {
    type: String,
    required: true,
  },
  experience: {
    type: Number,
    required: true,
  },
  certifications: [String],
  status: {
    type: String,
    enum: ["pending", "approved", "rejected"],
    default: "pending",
  },
  rating: {
    type: Number,
    default: 0,
  },
  totalReviews: {
    type: Number,
    default: 0,
  },
  totalEarnings: {
    type: Number,
    default: 0,
  },
  profileImage: {
    type: String,
    default: null,
  },
  createdAt: {
    type: Date,
    default: Date.now,
  },
  updatedAt: {
    type: Date,
    default: Date.now,
  },
});

export default mongoose.model("Provider", providerSchema);

File: Backend-SpringBoot/src/main/java/com/servesync/model/Provider.java
package com.servesync.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.springframework.data.annotation.CreatedDate;
import org.springframework.data.annotation.Id;
import org.springframework.data.annotation.LastModifiedDate;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "providers")
public class Provider {
    
    @Id
    private String id;
    
    @DBRef
    private User user;
    
    private String userId;
    
    private String businessName;
    
    private String description;
    
    private String category;
    
    private Integer experience;
    
    private List<String> certifications;
    
    private ProviderStatus status = ProviderStatus.PENDING;
    
    private Double rating = 0.0;
    
    private Integer totalReviews = 0;
    
    private Double totalEarnings = 0.0;
    
    private String profileImage;
    
    @CreatedDate
    private LocalDateTime createdAt;
    
    @LastModifiedDate
    private LocalDateTime updatedAt;
    
    public enum ProviderStatus {
        PENDING, APPROVED, REJECTED;
        
        @com.fasterxml.jackson.annotation.JsonCreator
        public static ProviderStatus fromString(String value) {
            return ProviderStatus.valueOf(value.toUpperCase());
        }
    }
}
